bundles/org.openhab.ui/web/src/assets/definitions/blockly/utils.js

```diff
@@ -44,19 +44,27 @@ export function addDateSupport () {
 
   let getzdt = Blockly.JavaScript.provideFunction_(
     'getZonedDateTime', [
-      '/* Try to detect the format based on its length */',
       'function ' + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ + '(datetime) {',
       '  datetime = String(datetime).replace(\'T\', \' \')',
-      '  switch (datetime.length) {',
-      `    case 10: return ${zdt}.parse(datetime + ' 00:00:00+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
-      `    case 16: return ${zdt}.parse(datetime + ':00+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
-      `    case 19: return ${zdt}.parse(datetime + '+00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
-      `    case 25: return ${zdt}.parse(datetime, dtf.ofPattern('yyyy-MM-dd HH:mm:ssz'));`,
-      `    case 23: return ${zdt}.parse(datetime + ' +00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ss.SSS z'));`,
-      `    case 26: return ${zdt}.parse(datetime + ' +00:00', dtf.ofPattern('yyyy-MM-dd HH:mm:ss.SSSSSS z'));`,
-      `    case 29: return ${zdt}.parse(datetime, dtf.ofPattern('yyyy-MM-dd HH:mm:ss.SSSSz'));`,
-      `    case 32: return ${zdt}.parse(datetime, dtf.ofPattern('yyyy-MM-dd HH:mm:ss.SSSSSSSz'));`,
-      `    case 28: return ${zdt}.parse(datetime.slice(0,26) + ':' + datetime.slice(26,28), dtf.ofPattern('yyyy-MM-dd HH:mm:ss.SSSSz'));`,
+      '  var regex_date =             /^\\d{4}-\\d{2}-\\d{2}$/; ',
+      '  var regex_date_time_min =    /^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_sec =    /^\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_sec_tz = /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}[+-]\\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_ms =     /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}$/;   ',
+      '  var regex_date_time_us =     /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{6}$/;  ',
+      '  var regex_date_time_ms_tz =  /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}[+-]\\d{2}:\\d{2}$/; ',
+      '  var regex_date_time_us_tz =  /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{6}[+-]\\d{2}:\\d{2}$/; ',
+      '  var regex_oh =               /^\\d{4}-\\d{2}-\\d{2}[T ]\\d{2}:\\d{2}:\\d{2}\\.\\d{3}[+-]\\d{4}$/;   ',
```

Comment:
I don't suppose these dangling EOL spaces were intentional?

Reply:
Oh yes, definitely. I do not trust any regex right away ;-) 

No, the spaces weren't intended. I removed them.